user-service/infrastructure/mongo/client.go

```diff
@@ -0,0 +1,34 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// TODO: remove hardcoded url
+const connectionString = "mongodb://localhost:27017/"
+
+func InitMongoClient() (*mongo.Client, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	clientOptions := options.Client().ApplyURI(connectionString)
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect: %w", err)
+	}
+
+	if err = client.Ping(ctx, nil); err != nil {
+		err := client.Disconnect(context.Background())
+		if err != nil {
+			return nil, fmt.Errorf("failed to disconnect server: %w", err)
+		}
+		return nil, fmt.Errorf("failed to ping serve: %w", err)
+	}
```

Comment:
This will not return a correct error on failure of `client.Ping` due to the `err` variable being overwritten by the disconnect, on successful disconnect `err` becomes `nil`.

In cases of command `ping` already failing, we may be able to simply ignore the error of `client.Disconnect` to return the actual failed message to the console:

```suggestion
	if err = client.Ping(ctx, nil); err != nil {
		// Ignoring disconnect errors to prioritize `ping` failure
		_ = client.Disconnect(context.Background())
		return nil, fmt.Errorf("failed to ping serve: %w", err)
	}
```